Multi-interface support

jamilbk opened the pull request:
Fixes #549 

- [x] Create networks table with proper fields, validations
- [x] Create migration to move config-file settings into first row of `networks` table, add `network_id` to `devices`
- [ ] Add new UI section, forms, modal
- [ ] Update existing device form, fields to be network-aware
- [ ] Generating configs should be network-aware
- [ ] Propagate interface creation, configuration, and deletion to `fz_vpn`
- [ ] Propagate interface creation to `fz_wall`
- [x] Remove old WireGuard interface cruft from Chef recipes / scripts
- [x] Remove WireGuard interface config from config file
- [x] Add appropriate telemetry events
- [ ] Update docs

jamilbk committed:
Sanity check networks

jamilbk committed:
More test

Signed-off-by: Jamil Bou Kheir <jamilbk@users.noreply.github.com>

jamilbk committed:
checkpoint

princemaple commented on apps/fz_http/priv/repo/migrations/20220527025915_create_networks.exs:
```diff
@@ -0,0 +1,35 @@
+
+    create unique_index(:networks, [:ipv4_address], where: "ipv4_address IS NOT NULL")
+    create unique_index(:networks, [:ipv6_address], where: "ipv6_address IS NOT NULL")
+    create unique_index(:networks, [:interface_name])
+    create unique_index(:networks, [:private_key])
+    create unique_index(:networks, [:listen_port])
```
Do `interface_name` and `listen_port` need to take `site` / `user` into consideration? I probably don't understand the problem we are solving enough.

princemaple replied:
If it's per machine, it's all good here.

jamilbk replied:
That’s a good point. We don’t have multi site yet but it’s planned. I’ll add a `site_id` field to scope it.

jamilbk committed:
Bump npm

jamilbk committed:
Bump deps for dependabot issues

jamilbk committed:
Bump lockfile

jamilbk committed:
Checkpoint updated migration, refactoring

princemaple committed:
MFA (#660)

* 2fa wip

add encrypted map

gen schema

add nimble_totp

add module doc

add qrusty

bump base to enable use of qrusty

requires a newer version of glibc

Revert "add qrusty"

This reverts commit 27eaf7864ea98578186c53ed4c5f650e244e8ede.

cannot be used on bionic

add eqrcode

scaffold

extract renderQR

Revert "add eqrcode"

This reverts commit fe6f74f18f13dfcc92dbdace657d194e5536e164.

use frontend qr code generation

display MFA methods

add description

* wip

* finish mfa totp addition workflow

* use real user

* allow deleting authenticator

* use valid? and :since

* make credo happy

* use mfa

* make credo happy

* add account page for unprivileged users

* eof

* add unit tests

* add integration tests
